viam-cartographer_internal_test.go

```diff
@@ -0,0 +1,348 @@
+// Package viamcartographer implements simultaneous localization and mapping.
+// This is an Experimental package.
+package viamcartographer
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	commonv1 "go.viam.com/api/common/v1"
+	v1 "go.viam.com/api/service/slam/v1"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/test"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var (
+	chunkSizePointCloudMap     int
+	chunkSizeInternalState     int
+	inputComponentRef          string
+	expectedPointCloudMapBytes []byte
+	expectedInternalStateBytes []byte
+	inputQuat                  map[string]interface{}
+	inputPose                  commonv1.Pose
```

Comment:
[q] Is there any chance that the tests below can be executed in parallel? If so, might it be better to define `inputComponentRef`, `inputQuat`, and `inputPose` in each test separately?